apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/status/AnnotationMatchExceptionCheckStrategy.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.agent.core.context.status;
+
+import java.lang.annotation.Annotation;
+import java.util.HashSet;
+import java.util.Objects;
+import org.apache.skywalking.apm.agent.core.boot.ServiceManager;
+
+public class AnnotationMatchExceptionCheckStrategy implements ExceptionCheckStrategy {
+
+    private final HashSet<Class<? extends Throwable>> ignoredExceptions = new HashSet<>(16);
+
+    @Override
+    public boolean isError(final Throwable e) {
+        if (ignoredExceptions.contains(e.getClass())) {
+            return false;
+        }
+        StatusCheckService statusTriggerService = ServiceManager.INSTANCE.findService(StatusCheckService.class);
+        Class<? extends Annotation> ignoredExceptionAnnotationClass = statusTriggerService.ignoredExceptionAnnotationClass;
```

Comment:
Accessing the field directly is not allowed.

Reply:
package access

Reply:
The same thing, all accesses should be through methods unless it is in the same class and children. But this is not the key, please check comments about the `annotation`, which is my primary concern.

Reply:
> The same thing, all accesses should be through methods unless it is in the same class and children. But this is

already reply